cmd/src/campaigns_new.go

```diff
@@ -0,0 +1,131 @@
+required fields.
+
+Usage:
+
+    src campaigns new [-f FILE]
+
+Examples:
+
+
+    $ src campaigns new -f campaign.spec.yaml
+
+`
+
+	flagSet := flag.NewFlagSet("new", flag.ExitOnError)
+
+	var (
+		fileFlag = flagSet.String("f", "campaign.yaml", "The name of campaign spec file to create.")
+	)
+
+	handler := func(args []string) error {
+		if err := flagSet.Parse(args); err != nil {
+			return err
+		}
+
+		if _, err := os.Stat(*fileFlag); !os.IsNotExist(err) {
+			return fmt.Errorf("file %s already exists", *fileFlag)
+		}
+
+		f, err := os.Create(*fileFlag)
+		if err != nil {
+			return errors.Wrapf(err, "failed to create file %s", *fileFlag)
+		}
+		defer f.Close()
+
+		tmpl, err := template.New("").Parse(campaignSpecTmpl)
+		if err != nil {
+			return err
+		}
+
+		author := campaigns.GitCommitAuthor{
+			Name:  "Sourcegraph",
+			Email: "campaigns@sourcegraph.com",
+		}
+
+		// Try to get better default values from git, ignore any errors.
+		if err := checkExecutable("git", "version"); err == nil {
+			gitAuthorName, err := getGitConfig("user.name")
+			if err == nil && gitAuthorName != "" {
+				author.Name = gitAuthorName
+			}
+
+			gitAuthorEmail, err := getGitConfig("user.email")
+			if err == nil && gitAuthorEmail != "" {
+				author.Email = gitAuthorEmail
+			}
+		}
+
+		err = tmpl.Execute(f, map[string]interface{}{"Author": author})
+		if err != nil {
+			return errors.Wrap(err, "failed to write campaign spec to file")
+		}
+
+		fmt.Printf("%s created.\n", *fileFlag)
+		return nil
+	}
+
+	campaignsCommands = append(campaignsCommands, &command{
+		flagSet: flagSet,
+		aliases: []string{},
+		handler: handler,
+		usageFunc: func() {
+			fmt.Fprintf(flag.CommandLine.Output(), "Usage of 'src campaigns %s':\n", flagSet.Name())
+			flagSet.PrintDefaults()
+			fmt.Println(usage)
+		},
+	})
+}
+
+func getGitConfig(attribute string) (string, error) {
+	cmd := exec.Command("git", "config", "--get", attribute)
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimSpace(string(out)), nil
+}
+
+const campaignSpecTmpl = `name: hello-world-campaign
+description: This campaign adds Hello World to READMEs
+
+# Find all repositories that contain a README.md file.
+on:
+  - repositoriesMatchingQuery: file:README.md
+
+# In each repository, run this command. Each repository's resulting diff is captured.
+steps:
+  - run: echo "Hello World" | tee -a $(find -name README.md)
+    container: alpine:3
+
+# Describe the changeset (e.g., GitHub pull request) you want for each repository.
+changesetTemplate:
+  title: Hello World
+  body: This adds Hello World to the README
+  branch: campaigns/hello-world # Push the commit to this branch.
+  commit:
+    author:
+      name: {{ .Author.Name }}
+      email: {{ .Author.Email }}
+    message: Append Hello World to all README.md files
+
+  # Change published to true once you're ready to create changesets on the code host.
+  published: false
+`
```

Comment:
I wonder if this spec is too opinionated: I know it's basically a copy of the hello world campaign in our docs, but I worry that it's not obvious enough where users need to dig in and change things, and if we'd be better served putting more obvious placeholders in instead for things like the name and description.

What do you think?

Reply:
I've been wondering about that too. What kept me from doing that was the fear that using meta-values like `run: <your shell script here>` or `name: campaign-name` makes it hard to learn what should go in the field (is `your shell script here` supposed to be a path? the name of a script? or a line?). I think with the values we have here it's obvious what's what and what needs to be changed. 

But I'll try something.